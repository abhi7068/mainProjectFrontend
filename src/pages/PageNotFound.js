import React, { Component } from 'react';

export default class PageNotFound extends Component {
  render() {
    return (
      <div>
        <h1 data-testid="h1">Error 404</h1>
        <p data-testid="p"> Page not Found</p>
      </div>
    );
  }
}
