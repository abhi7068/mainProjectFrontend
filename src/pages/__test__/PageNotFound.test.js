import React from 'react';
import PageNotFound from '../PageNotFound';
import { render } from '@testing-library/react';

it('it renders', () => {
  render(<PageNotFound />);
});

it('should take a snapshot', () => {
  const { asFragment } = render(<PageNotFound />);
  expect(asFragment(render(<PageNotFound />))).toMatchSnapshot();
});
it('it renders with correct text', () => {
  const { getByTestId } = render(<PageNotFound />);
  expect(getByTestId('h1')).toHaveTextContent('Error 404');
});
it('it renders with correct text', () => {
  const { getByTestId } = render(<PageNotFound />);
  expect(getByTestId('p')).toHaveTextContent('Page not Found');
});
