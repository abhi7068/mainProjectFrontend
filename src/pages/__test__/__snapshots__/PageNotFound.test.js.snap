// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="h1"
    >
      Error 404
    </h1>
    <p
      data-testid="p"
    >
       Page not Found
    </p>
  </div>
</DocumentFragment>
`;
